### DISCUSSION TITLE
po-files included but not specified as data files

### DISCUSSION THREAD

**Discussion by buhtz** [pypa/setuptools#4120] [Q&A] [answered]
I try to understand why po-files are included and installed in my package but I did not specified them as data files.

I have a custom build step that do use "msgfmt" to compile the po-files into mo-files. This seems to work. And this mo-files are specified as data-files and they are included.
But the po-files also. And that is what I don't want. 

The original project is a technical demo and can be found [here](https://codeberg.org/buhtz/tech-demo-python-packaging/src/branch/main/03_i18n_helloworld).

That is the original tree:
```
├── pyproject.toml
├── README.md
├── setup.py
├── src
│   └── helloworld-int
│       ├── __init__.py
│       ├── __main__.py
│       └── po
│           ├── de.po
│           ├── jp.po
│           └── messages.pot
└── tests
    ├── __init__.py
    └── test_dummy.py
```

That is the `pyproject.toml` where you can see that only the `mo`-files (generated in a custom build step) are included as data-files.
```
[build-system]
requires = [
    'setuptools',
]
build-backend = "setuptools.build_meta"

[project]
name = "helloworld-international"
version = "0.0.1"
description = """A demo translating Hello World in a GUI window."""
requires-python = ">=3.8"

dependencies = [
    "PySimpleGUI",
]

authors = [
    {name = "Christian Buhtz", email = "c.buhtz@posteo.jp"}
]

[project.urls]
homepage = "https://codeberg.org/buhtz/tech-demo-python-packaging"
repository = "https://codeberg.org/buhtz/tech-demo-python-packaging"

[project.scripts]
helloworldint = "helloworldint.__main__:main"

[tool.setuptools.package-dir]
helloworldint = 'src/helloworld-int'

[tool.setuptools.package-data]
# Directories not allowed. Files only.
helloworldint = [
    'locales/*/LC_MESSAGES/*.mo',
]
```

Here is some verbose output from `pip install .`
```
  adding 'helloworldint/__init__.py'
  adding 'helloworldint/__main__.py'
  adding 'helloworldint/locales/de/LC_MESSAGES/helloworldint.mo'
  adding 'helloworldint/locales/jp/LC_MESSAGES/helloworldint.mo'
  adding 'helloworldint/po/de.po'
  adding 'helloworldint/po/jp.po'
  adding 'helloworld_international-0.0.1.dist-info/METADATA'
  adding 'helloworld_international-0.0.1.dist-info/WHEEL'
  adding 'helloworld_international-0.0.1.dist-info/entry_points.txt'
  adding 'helloworld_international-0.0.1.dist-info/top_level.txt'
  adding 'helloworld_international-0.0.1.dist-info/RECORD'
```

**Comment by abravalheri** [comment, marked as answer]
I suspect this is probably because `tool.setuptools.include-package-data` default value in `pyproject.toml` is true. Isn't it the case?

When we were designing the support for `pyproject.toml` the feedback we received is that it was better to change the default of `include-package-data` to True because that is what most beginners would expect.
This and other changes in behaviour were made in order to steer the usage of `pyproject.toml` towards "convention over configuration".

I recently updated the https://setuptools.pypa.io/en/latest/userguide/datafiles.html docs to make this more evident. I hope that helps.
